File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java



class BirdWatcher {
    private final int[] birdsPerDay;
    private final int[] lastWeek = {0,2,5,3,7,8,4}; 

    public BirdWatcher(int [] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return lastWeek;
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length-1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length-1] = getToday() + 1; 
    }

    public boolean hasDayWithoutBirds() {
        boolean zeroBirds = false;
        for(int countsOfDay : birdsPerDay){
            if(countsOfDay == 0){
                zeroBirds = true;
                break;
            }
        }
        return zeroBirds;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int totalBirdsCount = 0;
        try {
            // if number of days is zero then do not decrement value
            numberOfDays = numberOfDays == 0 ? 0: numberOfDays -1;
            for(int i=0; i<=numberOfDays ; i++){
                totalBirdsCount = totalBirdsCount + birdsPerDay[i];
            }
        } catch (Exception IndexOutOfBoundsException) {
            System.err.println("Week only has 7 days");
        }
        return totalBirdsCount;
    }

    public int getBusyDays() {
        int busyDays = 0;
        for(int dayCount : birdsPerDay){
            if(dayCount>=5){
                busyDays = busyDays+1;
            }
        }
        return busyDays;
    }
}

File: solutions/csharp/bird-watcher/4/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay = new int [] {2,5,0,7,4,1};

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek()
    {
        int[] birdCountLastWeek = new int [] { 0, 2, 5, 3, 7, 8, 4 };
        return birdCountLastWeek;
    }

    public int Today()
    {
        return birdsPerDay[birdsPerDay.Lenght - 1];
    }

    public void IncrementTodaysCount()
    {
        int newCount = birdsPerDay[birdsPerDayLenght - 1] + 1;
        birdsPerDay[birdsPerDayLenght - 1] = newCount;
    }

    public bool HasDayWithoutBirds()
    {
        foreach (int i in birdCountLastWeek)
        {
            if (i == 0)
            {
                return true;
            }
        }
        return false;
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int total = 0;
        for (int i = 0; i < numberOfDays; i++)
        {
            total = total + birdsPerDay[i];
        }
        return total;
    }

    public int BusyDays()
    {
        int busydays = 0;
        foreach (int i in birdsPerDay)
        {
            if (i >= 5)
            {
                busydays++;
            }
        }
        return busydays;
    }
}
